### PR TITLE
[GHA][MO][openvino-dev] Remove tests for legacy openvino-dev from GHA

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** We are removing openvino-dev in 2025.0 so all legacy tests are no longer needed in GHA.

**Ticket:** 157072

### CODE REVIEW

File: .github/workflows/job_onnx_models_tests.yml [hunk 1]
```diff
@@ -89,11 +89,6 @@ jobs:
 
           extras_to_install="onnx"
```
Comment (ilya-lavrenov) [inline .github/workflows/job_onnx_models_tests.yml:90]: drop this line as well

File: .github/workflows/job_python_unit_tests.yml [hunk 2]
```diff
@@ -102,17 +102,11 @@ jobs:
 
           extras_to_install="caffe,kaldi,onnx,tensorflow2,pytorch"
```
Comment (ilya-lavrenov) [inline .github/workflows/job_python_unit_tests.yml:103]: please, remove

File: .github/workflows/job_onnx_models_tests.yml (hunk 1 above)
Comment (rkazants) [reply .github/workflows/job_onnx_models_tests.yml]: Done

File: .github/workflows/windows_vs2019_release.yml [hunk 3]
```diff
@@ -309,9 +304,6 @@ jobs:
           # For torchvision to OpenVINO preprocessing converter
           python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/python/preprocess/torchvision/requirements.txt
 
-          # TODO: replace with Python API tests requirements
-          python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/mo/requirements_dev.txt
```
[the remark this answers is not in the document]
Comment (rkazants) [reply .github/workflows/windows_vs2019_release.yml:313]: Done, great catch!

File: .github/workflows/job_python_unit_tests.yml (hunk 2 above)
Comment (rkazants) [reply .github/workflows/job_python_unit_tests.yml]: great catch!

File: .github/workflows/job_python_unit_tests.yml [hunk 4]
```diff
@@ -100,19 +100,11 @@ jobs:
           # Install the core OV wheel
           python3 -m pip install ./openvino-*.whl
```
Comment (rkazants) [inline .github/workflows/job_python_unit_tests.yml:102]:
```suggestion

        working-directory: ${{ env.INSTALL_WHEELS_DIR }}
```

File: .github/workflows/windows_vs2019_release.yml [hunk 5]
```diff
@@ -296,11 +296,6 @@ jobs:
           $ovCoreWheelPath=Get-ChildItem -Path . -Filter openvino-*.whl | % { $_.FullName }
           python3 -m pip install "$ovCoreWheelPath"
```
Comment (rkazants) [inline .github/workflows/windows_vs2019_release.yml:298]:
```suggestion

        working-directory: ${{ env.INSTALL_WHEELS_DIR }}
```

File: .github/workflows/windows_vs2019_release.yml [hunk 6]
```diff
@@ -296,11 +296,7 @@ jobs:
...
 
-          # Find and install the dev OV wheel
-          $ovDevWheelPath=Get-ChildItem -Path . -Filter openvino_dev*.whl | % { $_.FullName }
-          python3 -m pip install "$ovDevWheelPath[caffe,kaldi,onnx,tensorflow2,pytorch]"
         working-directory: ${{ env.INSTALL_WHEELS_DIR }}
```
Comment (rkazants) [inline .github/workflows/windows_vs2019_release.yml:299]:
```suggestion
        working-directory: ${{ env.INSTALL_WHEELS_DIR }}

```

File: .github/workflows/job_onnx_models_tests.yml [hunk 7]
```diff
@@ -87,13 +87,6 @@ jobs:
           # Install the core OV wheel
           python3 -m pip install ./openvino-*.whl
```
Comment (rkazants) [inline .github/workflows/job_onnx_models_tests.yml:89]:
```suggestion
        working-directory: ${{ env.INSTALL_WHEELS_DIR }}

```

File: tests/layer_tests/ovc_python_api_tests/test_pytorch.py [hunk 8]
```diff
@@ -366,188 +366,6 @@ def scripted_fn(x: torch.Tensor, y: torch.Tensor):
     return scripted_fn, ref_model, {'input': [(inp_shape, Type.f32), (inp_shape, Type.f32)]}
```
Comment (rkazants) [inline tests/layer_tests/ovc_python_api_tests/test_pytorch.py:368]:
```suggestion
def create_pytorch_nn_module_layout_list(tmp_dir):
    from openvino.runtime import Layout
    pt_model = make_pt_model_two_inputs()
    shape = [1, 3, 10, 10]

    shape = PartialShape(shape)
    ref_model = make_ref_pt_model_two_inputs(shape)
    ref_model.inputs[0].node.layout = Layout('nchw')
    ref_model.inputs[1].node.layout = Layout('nhwc')

    return pt_model, ref_model, {
        'input': [(shape, np.float32), (shape, np.float32)], 'layout': ['nchw', Layout('nhwc')],
        'use_convert_model_from_mo': True
    }


def create_pytorch_nn_module_layout_list_case2(tmp_dir):
    from openvino.runtime import Layout
    pt_model = make_pt_model_two_inputs()
    shape = [1, 3, 10, 10]

    shape = PartialShape(shape)
    ref_model = make_ref_pt_model_two_inputs(shape)
    ref_model.inputs[0].node.layout = Layout('nchw')
    ref_model.inputs[1].node.layout = Layout('nhwc')

    return pt_model, ref_model, {
        'input': [(shape, np.float32), (shape, np.float32)], 'layout': ('nchw', Layout('nhwc')),
        'use_convert_model_from_mo': True}


def create_pytorch_nn_module_mean_list_compression_disabled(tmp_dir):
    pt_model = make_pt_model_two_inputs()
    shape = [1, 10, 10, 3]

    shape = PartialShape(shape)
    param1 = ov.opset8.parameter(shape)
    param2 = ov.opset8.parameter(shape)
    const1 = ov.opset8.constant([[[[-0.0, -0.0, -0.0]]]], dtype=np.float32)
    const2 = ov.opset8.constant([[[[-0.0, -0.0, -0.0]]]], dtype=np.float32)
    add1 = ov.opset8.add(param1, const1)
    add2 = ov.opset8.add(param2, const2)
    mul = ov.opset8.multiply(add1, add2)
    relu = ov.opset8.relu(mul)
    sigm = ov.opset8.sigmoid(relu)

    parameter_list = [param1, param2]
    ref_model = Model([sigm], parameter_list, "test")

    return pt_model, ref_model, {'input': [(shape, np.float32), (shape, np.float32)],
                                 'mean_values': [[0, 0, 0], [0, 0, 0]],
                                 'compress_to_fp16': False, 'use_convert_model_from_mo': True}


def create_pytorch_nn_module_mean_list_compression_default(tmp_dir):
    # when 'use_convert_model_from_mo': True by default compression in convert_model is disabled
    # therefore decompression Converts will not be present
    pt_model = make_pt_model_two_inputs()
    shape = [1, 10, 10, 3]

    shape = PartialShape(shape)
    param1 = ov.opset8.parameter(shape)
    param2 = ov.opset8.parameter(shape)
    const1 = ov.opset8.constant([[[[-0.0, -0.0, -0.0]]]], dtype=np.float32)
    const2 = ov.opset8.constant([[[[-0.0, -0.0, -0.0]]]], dtype=np.float32)
    add1 = ov.opset8.add(param1, const1)
    add2 = ov.opset8.add(param2, const2)
    mul = ov.opset8.multiply(add1, add2)
    relu = ov.opset8.relu(mul)
    sigm = ov.opset8.sigmoid(relu)

    parameter_list = [param1, param2]
    ref_model = Model([sigm], parameter_list, "test")

    return pt_model, ref_model, {'input': [(shape, np.float32), (shape, np.float32)],
                                 'mean_values': [[0, 0, 0], [0, 0, 0]],
                                 'use_convert_model_from_mo': True}


def create_pytorch_nn_module_mean_list_compression_enabled(tmp_dir):
    pt_model = make_pt_model_two_inputs()
    shape = [1, 10, 10, 3]

    shape = PartialShape(shape)
    param1 = ov.opset8.parameter(shape)
    param2 = ov.opset8.parameter(shape)
    const1 = ov.opset8.constant([[[[-0.0, -0.0, -0.0]]]], dtype=np.float16)
    const2 = ov.opset8.constant([[[[-0.0, -0.0, -0.0]]]], dtype=np.float16)
    const1_decompressed = ov.opset8.convert(
        const1, destination_type=np.float32)
    const2_decompressed = ov.opset8.convert(
        const2, destination_type=np.float32)

    add1 = ov.opset8.add(param1, const1_decompressed)
    add2 = ov.opset8.add(param2, const2_decompressed)
    mul = ov.opset8.multiply(add1, add2)
    relu = ov.opset8.relu(mul)
    sigm = ov.opset8.sigmoid(relu)

    parameter_list = [param1, param2]
    ref_model = Model([sigm], parameter_list, "test")

    return pt_model, ref_model, {
        'input': [(shape, np.float32), (shape, np.float32)], 'mean_values': [[0, 0, 0], [0, 0, 0]],
        'compress_to_fp16': True, 'use_convert_model_from_mo': True}


def create_pytorch_nn_module_scale_list_compression_disabled(tmp_dir):
    pt_model = make_pt_model_two_inputs()
    shape = [1, 10, 10, 3]

    shape = PartialShape(shape)
    param1 = ov.opset8.parameter(shape)
    param2 = ov.opset8.parameter(shape)
    const1 = ov.opset8.constant([[[[1, 1, 1]]]], dtype=np.float32)
    const2 = ov.opset8.constant([[[[1, 1, 1]]]], dtype=np.float32)
    sub1 = ov.opset8.multiply(param1, const1)
    sub2 = ov.opset8.multiply(param2, const2)
    mul = ov.opset8.multiply(sub1, sub2)
    relu = ov.opset8.relu(mul)
    sigm = ov.opset8.sigmoid(relu)

    parameter_list = [param1, param2]
    ref_model = Model([sigm], parameter_list, "test")

    return pt_model, ref_model, {'input': [(shape, np.float32), (shape, np.float32)],
                                 'scale_values': [[1, 1, 1], [1, 1, 1]],
                                 'compress_to_fp16': False, 'use_convert_model_from_mo': True}


def create_pytorch_nn_module_scale_list_compression_default(tmp_dir):
    # when 'use_convert_model_from_mo': True by default compression in convert_model is disabled
    # therefore decompression Converts will not be present
    pt_model = make_pt_model_two_inputs()
    shape = [1, 10, 10, 3]

    shape = PartialShape(shape)
    param1 = ov.opset8.parameter(shape)
    param2 = ov.opset8.parameter(shape)
    const1 = ov.opset8.constant([[[[1, 1, 1]]]], dtype=np.float32)
    const2 = ov.opset8.constant([[[[1, 1, 1]]]], dtype=np.float32)
    sub1 = ov.opset8.multiply(param1, const1)
    sub2 = ov.opset8.multiply(param2, const2)
    mul = ov.opset8.multiply(sub1, sub2)
    relu = ov.opset8.relu(mul)
    sigm = ov.opset8.sigmoid(relu)

    parameter_list = [param1, param2]
    ref_model = Model([sigm], parameter_list, "test")

    return pt_model, ref_model, {'input': [(shape, np.float32), (shape, np.float32)],
                                 'scale_values': [[1, 1, 1], [1, 1, 1]],
                                 'use_convert_model_from_mo': True}


def create_pytorch_nn_module_scale_list_compression_enabled(tmp_dir):
    pt_model = make_pt_model_two_inputs()
    shape = [1, 10, 10, 3]

    shape = PartialShape(shape)
    param1 = ov.opset8.parameter(shape)
    param2 = ov.opset8.parameter(shape)
    const1 = ov.opset8.constant([[[[1, 1, 1]]]], dtype=np.float16)
    const1_decompressed = ov.opset8.convert(
        const1, destination_type=np.float32)
    const2 = ov.opset8.constant([[[[1, 1, 1]]]], dtype=np.float16)
    const2_decompressed = ov.opset8.convert(
        const2, destination_type=np.float32)
    mul1 = ov.opset8.multiply(param1, const1_decompressed)
    mul2 = ov.opset8.multiply(param2, const2_decompressed)
    mul3 = ov.opset8.multiply(mul1, mul2)
    relu = ov.opset8.relu(mul3)
    sigm = ov.opset8.sigmoid(relu)

    parameter_list = [param1, param2]
    ref_model = Model([sigm], parameter_list, "test")

    return pt_model, ref_model, {'input': [(shape, np.float32), (shape, np.float32)],
                                 'scale_values': [[1, 1, 1], [1, 1, 1]],
                                 'compress_to_fp16': True, 'use_convert_model_from_mo': True}
```

File: tests/layer_tests/ovc_python_api_tests/test_pytorch.py [hunk 9]
```diff
@@ -546,8 +545,6 @@ def create_pytorch_nn_module_scale_list_compression_enabled(tmp_dir):
     return pt_model, ref_model, {'input': [(shape, np.float32), (shape, np.float32)],
                                  'scale_values': [[1, 1, 1], [1, 1, 1]],
                                  'compress_to_fp16': True, 'use_convert_model_from_mo': True}
```
Comment (rkazants) [inline tests/layer_tests/ovc_python_api_tests/test_pytorch.py:547]:
```suggestion
                                 'compress_to_fp16': True, 'use_convert_model_from_mo': True}


```

File: tests/layer_tests/requirements.txt [hunk 10]
```diff
@@ -1,6 +1,7 @@
 -c ../constraints.txt
 # paddlepaddle #  ticket 95904
 numpy
+onnx
```
Comment (rkazants) [inline tests/layer_tests/requirements.txt:4]:
```suggestion
```

File: tests/constraints.txt [hunk 11]
```diff
@@ -13,6 +13,7 @@ defusedxml>=0.7.1
 fastjsonschema~=2.17.1
 tensorflow>=2.5,<2.19.0
 requests>=2.25.1
+onnx>=1.8.1,<=1.17.0
```
Comment (rkazants) [inline tests/constraints.txt:16]:
```suggestion
```

File: .github/workflows/windows_vs2019_release.yml [hunk 12]
```diff
@@ -309,43 +306,26 @@ jobs:
...
-          # TODO: replace with Python API tests requirements
-          python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/mo/requirements_dev.txt
+          # For validation of Python API
+          python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/bindings/python/requirements_test.txt
 
           # For getting rid of SSL issues during model downloading for unit tests
           python3 -m pip install certifi
 
       - name: Set SSL_CERT_FILE for model downloading for unit tests
         run: echo SSL_CERT_FILE=$(python3 -m certifi) >> $env:GITHUB_ENV
```
Comment (rkazants) [inline .github/workflows/windows_vs2019_release.yml:317]:
```suggestion

      - name: Python API Tests
        #if: fromJSON(needs.smart_ci.outputs.affected_components).Python_API.test # Ticket: 127101
        shell: cmd
        run: |
          set PYTHONPATH=${{ env.LAYER_TESTS_INSTALL_DIR }};%PYTHONPATH%
          python3 -m pytest -sv ${{ env.INSTALL_TEST_DIR }}/pyopenvino ${{ env.PYTHON_STATIC_ARGS }} --junitxml=${{ env.INSTALL_TEST_DIR }}/TEST-Pyngraph.xml --ignore=${{ env.INSTALL_TEST_DIR }}/pyopenvino/tests/test_utils/test_utils.py

```

File: .github/workflows/job_python_unit_tests.yml [hunk 13]
```diff
@@ -91,28 +91,31 @@ jobs:
...
+          # For torchvision to OpenVINO preprocessing converter
+          python3 -m pip install -r ${INSTALL_TEST_DIR}/python/preprocess/torchvision/requirements.txt
+
+          # layer test requirements
+          python3 -m pip install -r ${LAYER_TESTS_INSTALL_DIR}/requirements.txt
+
+      - name: Install ONNX tests dependencies
+        run: |
+          #  ONNX tests requirements
+          python3 -m pip install -r ${LAYER_TESTS_INSTALL_DIR}/requirements_onnx
```
Comment (rkazants) [inline .github/workflows/job_python_unit_tests.yml:118]:
```suggestion
          python3 -m pip install -r ${INSTALL_TEST_DIR}/requirements_onnx
```

File: .github/workflows/windows_vs2019_release.yml [hunk 14]
```diff
@@ -309,43 +306,26 @@ jobs:
           # For torchvision to OpenVINO preprocessing converter
           python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/python/preprocess/torchvision/requirements.txt
 
-          # TODO: replace with Python API tests requirements
-          python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/mo/requirements_dev.txt
+          # For validation of Python API
+          python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/bindings/python/requirements_test.txt
```
Comment (rkazants) [inline .github/workflows/windows_vs2019_release.yml:310]:
```suggestion
          python3 -m pip install -r ${{ env.INSTALL_TEST_DIR }}/bindings/python/requirements_test.txt

          #  ONNX tests requirements
          python3 -m pip install -r ${INSTALL_TEST_DIR}/requirements_onnx
```

File: .github/dependabot.yml [hunk 15]
```diff
@@ -41,7 +41,7 @@ updates:
       - "rkazants"
     versioning-strategy: increase-if-necessary
 
-  # Model Optimizer, openvino_dev and Benchmark tool
+  # ovc tool
```
Comment (akuporos) [inline .github/dependabot.yml:44]:
```suggestion
  # ovc and Benchmark tools
```

File: tests/CMakeLists.txt [hunk 16]
```diff
@@ -9,3 +9,4 @@ add_subdirectory(e2e_tests)
 
 install(FILES requirements_pytorch DESTINATION tests COMPONENT tests EXCLUDE_FROM_ALL)
 install(FILES requirements_tensorflow DESTINATION tests COMPONENT tests EXCLUDE_FROM_ALL)
+install(FILES requirements_onnx DESTINATION tests COMPONENT tests EXCLUDE_FROM_ALL)
```
Comment (ilya-lavrenov) [inline tests/CMakeLists.txt:12]: we can merge 3 `install` calls with:
```cmake
install(FILES requirements_pytorch requirements_tensorflow requirements_onnx
        DESTINATION tests COMPONENT tests EXCLUDE_FROM_ALL)
```

**Review by ilya-lavrenov — APPROVED** [review]

File: tests/CMakeLists.txt (hunk 16 above)
Comment (rkazants) [reply tests/CMakeLists.txt]: Done

**Review by gkrivor — APPROVED** [review]
Hi, did you check this file https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/ci_utils/onnxruntime/version is it still required?

**Review by akashchi — APPROVED** [review]
GHA changes LGTM.

**Comment by rkazants** [comment]
> Hi, did you check this file https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/ci_utils/onnxruntime/version is it still required?

No, I didn't. I think it can be a separate story to adjust files for ONNX FE validation. Anyway these tests will stay for new ONNX FE and does not relate legacy story.

**Review by popovaan — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

**Pull request merged**